test/built-ins/Symbol/desc-to-string.js

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-symbol-constructor
+es6id: 19.4.1
+description: The first argument is coerced to a String value
+info: |
+    1. If NewTarget is not undefined, throw a TypeError exception.
+    2. If description is undefined, let descString be undefined.
+    2. Else, let descString be ? ToString(description).
+    3. Return a new unique Symbol value whose [[Description]] value is
+       descString.
+---*/
+
+var calls, val;
+
+val = {
+  toString: function() {
+    calls += 'toString';
+    return {};
+  },
+  valueOf: function() {
+    calls += 'valueOf';
+  }
+};
+
+calls = '';
+Symbol(val);
+assert.sameValue(calls, 'toStringvalueOf');
+
+val = {
+  toString: function() {
+    calls += 'toString';
+  },
+  valueOf: function() {
+    calls += 'valueOf';
+  }
+};
+
+calls = '';
+Symbol(val);
+assert.sameValue(calls, 'toString');
+
+val = {
+  toString: null,
+  valueOf: function() {
+    calls += 'valueOf';
+  }
+};
+
+calls = '';
+Symbol(val);
+assert.sameValue(calls, 'valueOf');
+
+val = {
+  toString: null,
+  valueOf: function() {
+    calls += 'valueOf';
+    return {};
+  }
+};
+
+calls = '';
+assert.throws(TypeError, function() {
+  Symbol(val);
+});
+assert.sameValue(calls, 'valueOf');
+
+val = {
+  toString: function() {
+    calls += 'toString';
+    return {};
+  },
+  valueOf: function() {
+    calls += 'valueOf';
+    return {};
+  }
+};
+
+calls = '';
+assert.throws(TypeError, function() {
+  Symbol(val);
+});
+assert.sameValue(calls, 'toStringvalueOf');
+
+assert.throws(TypeError, function() {
+  Symbol(Symbol('1'));
+});
```

Comment:
it would be good to add a message to these throws assertions, this way it's easier to keep track what exactly failed.

Also, this last assertion seems isolated to the other cases, can we have it in an extra file? It's a good justification for the fact it's not possible to make a Symbol from another Symbol.